src/FESpaces/FESpacesWithDofPotentiallyRemoved.jl

```diff
@@ -0,0 +1,185 @@
+    new{CS,RD}(space,cs,rd,dof_to_remove)
+  end
+end
+
+# Genuine functions
+function num_free_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS}
+  num_free_dofs(f.space)-1
+end
+
+function num_free_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS}
+  num_free_dofs(f.space)
+end
+
+function zero_free_values(f::FESpaceWithDofPotentiallyRemoved)
+  zeros(num_free_dofs(f))
+end
+
+function get_cell_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS}
+  cell_dofs = get_cell_dofs(f.space)
+  CellDofsWithDofRemoved(cell_dofs,f.dof_to_remove)
+end
+
+function get_cell_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS}
+  get_cell_dofs(f.space)
+end
+
+
+num_dirichlet_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS} = 1
+
+num_dirichlet_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS} = 0
+
+function zero_dirichlet_values(f::FESpaceWithDofPotentiallyRemoved)
+  T = Float64 # TODO
+  zeros(T,num_dirichlet_dofs(f))
+end
+
+
+num_dirichlet_tags(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS} = 1
+
+num_dirichlet_tags(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS} = 0
+
+get_dirichlet_dof_tag(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS} = Int8[1,]
+
+get_dirichlet_dof_tag(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS} = Int8[]
+
+function scatter_free_and_dirichlet_values(
+  f::FESpaceWithDofPotentiallyRemoved{CS,true},fv,dv) where {CS}
+  @assert length(dv) == 1
+  _dv = similar(dv,eltype(dv),0)
+  _fv = vcat(view(fv,1:f.dof_to_remove-1),
+             dv,
+             view(fv,f.dof_to_remove:length(fv))) # TODO lazy append
+ scatter_free_and_dirichlet_values(f.space,_fv,_dv)
+end
+
+function scatter_free_and_dirichlet_values(
+  f::FESpaceWithDofPotentiallyRemoved{CS,false},fv,dv) where {CS}
+  @assert length(dv) == 0
+ scatter_free_and_dirichlet_values(f.space,fv,dv)
+end
+
+function gather_free_and_dirichlet_values(
+  f::FESpaceWithDofPotentiallyRemoved{CS,true},cv) where {CS}
+  _fv, _dv = gather_free_and_dirichlet_values(f.space,cv)
+  @assert length(_dv) == 0
+  fv = vcat(view(_fv,1:f.dof_to_remove-1),
+            view(_fv,f.dof_to_remove+1:length(_fv))) # TODO: can we avoid
+                                          # allocating new memory?
+  dv = view(_fv,f.dof_to_remove:f.dof_to_remove)
+  (fv, dv)
+end
+
+function gather_free_and_dirichlet_values(
+  f::FESpaceWithDofPotentiallyRemoved{CS,false},cv) where {CS}
+  gather_free_and_dirichlet_values(f.space,cv)
+end
+
+function gather_free_and_dirichlet_values!(
+  fv,dv,f::FESpaceWithDofPotentiallyRemoved{CS,true},cv) where {CS}
+  _fv, _dv = gather_free_and_dirichlet_values(f.space,cv)
+  @assert length(_dv) == 0
+  fv .= vcat(view(_fv,1:f.dof_to_remove-1),
+            view(_fv,f.dof_to_remove+1:length(_fv))) # TODO: can we avoid
+                                          # allocating new memory?
+  dv .= view(_fv,f.dof_to_remove:f.dof_to_remove)
+  (fv, dv)
+end
+
```

Comment:
In all these functions, we should decide which is the way to deal with the vector with one row eliminated in an efficient way. We could probably create a lazy vector doing this, using what we already have in Gridap. It is a quite obvious and would not involve any allocation instead of using views + vcat. Idem for the vector with only one touched value.

Reply:
Another option is to implement a `VectorWithIndexRemoved` in addition to the `VectorWithIndexInserted` I commented above.

Reply:
I will implement `VectorWithIndexRemoved` and `VectorWithIndexInserted` because it will be even more efficient than using lazy vectors, and can be useful for other scenarios as well. Besides, lazy vectors do not currently support write operations, broadcasted assignments, etc. Is there any fundamental reason behind this? or it is just because it is missing? I do not need it, anyway.

Reply:
Done in PR https://github.com/gridap/Gridap.jl/pull/401